### DISCUSSION TITLE
Merge of Union

### DISCUSSION THREAD

**Discussion by dmfilipenko** [millsp/ts-toolbelt#193] [Wiki]
#### Describe your question

What is the best way to transform this
```typescript
type A = {id: string} 
type B = {uuid: string} | {name: string}
```
to something like this
```typescript
type C = {
		id: string;
    	uuid: string;
	} | {
		id: string;
		name: string;
	}
```
#### Search tags, topics
merge unions

**Comment by millsp** [comment]
Hi @dmfilipenko, I just released an upgraded version of `Either` for you. Now you can do this:

```ts
import {O, U} from 'ts-toolbelt'

type A = {id: string}
type B = {uuid: string} | {name: string}

type t = O.Either<U.Merge<B | A>, 'id' | 'uuid', 0>
```

TypeScript does not perform excess property checks on unions, so I guess you'd rather use the first:

```ts
type t = O.Either<U.Merge<B | A>, 'id' | 'uuid'>
```

Cheers

**Comment by dmfilipenko** [comment]
@pirix-gh <3 thank you!
